tests/Browser/ProductDetailTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Tests\Browser;
+
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+use App\Models\Product;
+use App\Models\User;
+
+class ProductDetailTest extends DuskTestCase
+{
+    public function test_view_product_detail_english()
+    {
+        $this->browse(function ($browser) {
+            $product = Product::find(10);
+            $browser->loginAs(User::find(1))
```

Comment:
trình bày các function test trong file test này

Reply:
1. function test_view_product_detail_english()
test các text hiển thị với ngôn ngữ tiếng anh
 - line 16: lấy product có id = 10 để test trang ProductDetail
 - line 17: login với user có id = 1
 - line 18: đi tới trang ProductDetail
 - line 22: chọn ngôn ngữ là ENGLISH
 - line 30: di chuyển chuột vào tên người dùng để hiển thị modal
 - assertSee(): tìm kiếm text đó với trang ProductDetail

2. function  test_view_product_detail_vietnamese()
 - tương tự function (1), nhưng test text tiếng việt
 - vì đã login ở function (1), session login vẫn còn đó nên trong function này em không login lại nữa

3. function test_redirect_to_cart()
test icon cart chuyển hướng tới trang 'Giỏ hàng'
 - line 77: đi tới trang ProductDetail có id = 10
 - line 78: click vào icon cart
 - line 79: kiểm tra url sau khi click

4. function test_redirect_to_order_history()
test link chuyển hướng tới trang 'Lịch sủ mua hàng'
 - line 88: mouseover để hiển thị modal
 - line 90: click vào 'Lịch sử mua hàng'
 - line 91: kiểm tra url sau khi click
 - waitForText(): theo em hiểu thì, khi line 87 (và 89) chạy, view chưa hiển thị hết thì line 88 (và 90) đã chạy rồi, ,nên sẽ báo lỗi ko tìm thấy text đó. Vì vậy, cần hàm waitForText() (hoặc pause()) mục đích là để chờ dòng test trên nó chạy xong thì mới chạy dòng test kế tiếp (trong thời gian mặc định là 5s nếu vẫn ko thì thấy thì nó sẽ in ra 1 ngoại lệ)

5. function test_button_add_to_cart()
test button 'Add to cart'
 - line 100: chờ cho selector #add-to-cart xuất hiện
 - line 101: ấn chọn size đầu tiên của product
 - line 102: chọn số lượng là 1
 - line 103: click button add to cart
 - line 104: kiểm tra url sau khi click